db/error_handler.cc

```diff
@@ -204,6 +198,12 @@ Status ErrorHandler::SetBGError(const Status& bg_err, BackgroundErrorReason reas
 
   new_bg_err = Status(bg_err, sev);
 
+  // Check if recovery is currently in progress. If it is, we will save this
```

Comment:
Can we move this block just before if (new_bg_err == Status::NoSpace()) ?

Reply:
We could, but is there a particular reason? The reason for moving this down here is so recovery_error_ will have the correct severity. This way, the caller of DB::Resume() will receive an error indicating whether they can retry resume later or not. We don't necessarily care how new_bg_err_ is handled from this point on.

Reply:
Thanks for the explanation. Can you remove the ";" in line 210 in this PR?